beacon-chain/state/phase0.go

```diff
@@ -18,6 +18,12 @@ type BeaconState interface {
 	Copy() BeaconState
 	HashTreeRoot(ctx context.Context) ([32]byte, error)
 	FutureForkStub
+	StateProver
+}
+
+// StateProver defines the ability to create Merkle proofs for beacon state fields.
+type StateProver interface {
+	ProveFinalizedRoot() ([][]byte, error)
```

Comment:
Optional:

```suggestion
	FinalizedRootProofs() ([][]byte, error)
```

Reply:
It's a single proof, maybe I can just call it FinalizedRootProof, but Prove as a prefix is a bit easier to read and immediately tells you what it does

Reply:
All future proof methods can have the Prove prefix, so ProveNextSyncCommitte, etc.

Reply:
`ProveFinalizedRoot` makes me think of something like `ProveFinalizedRoot([][]byte) (bool, error)` given the prefix is a verb. I think this is intended to be a getter. No hard preference for either. You choose